samtranslator/model/__init__.py

```diff
@@ -433,6 +434,21 @@ def get_passthrough_resource_attributes(self) -> Dict[str, Any]:
                 attributes[resource_attribute] = self.resource_attributes.get(resource_attribute)
         return attributes
 
+    def assign_tags(self, tags: Dict[str, Any]) -> None:
+        """
+        Assigns tags to the resource. This function assumes that generated resources always have
+        the tags property called `Tags`
+
+        There are CFN resources that do not follow the assumption such as 'AWS::Backup::Framework'
+        We may need to modify the code to account for thia scenario in the future. We could do this
+        by function overriding this function in the class definition of that resource. An example
+        is provided in 'AWS::ApiGatewayV2::Api'
+
+        :param tags: Dictionary of tags to be assigned to the resource
```

Comment:
Maybe so it's a little clearer it's by design and not a limitation that needs to be fixed:

```suggestion
        Assigns tags to the resource. This function assumes that generated resources always have
        the tags property called `Tags` that takes a list of key-value objects.

        Override this function if the above assumptions do not apply to the resource (e.g. different
        property name or type (see e.g. 'AWS::ApiGatewayV2::Api').

        :param tags: Dictionary of tags to be assigned to the resource
```